src/main/java/dev/jbang/util/JavaUtil.java

```diff
@@ -6,22 +6,101 @@
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
-import java.util.Comparator;
-import java.util.Optional;
+import java.util.*;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 import java.util.stream.Stream;
 
 import javax.annotation.Nonnull;
 import javax.annotation.Nullable;
 
-import dev.jbang.net.JdkManager;
-import dev.jbang.net.JdkProvider;
+import dev.jbang.jvm.Jdk;
+import dev.jbang.jvm.JdkManager;
+import dev.jbang.jvm.JdkProvider;
+import dev.jbang.jvm.jdkproviders.*;
 
 public class JavaUtil {
 
 	private static Integer javaVersion;
 
+	private static JdkManager jdkManager = null;
+
+	@Nonnull
+	public static JdkManager jdkManager() {
```

Comment:
we seem to call this in many places - but how is going to take into affect jbang provided config of i.e. providers?

Reply:
Right now the idea is that the config gets used exactly in that file you're referencing. So the builder will use the config to initialize the jdk manager, including the providers. The jdk manager itself is then completely agnostic where its configuration comes from.

Reply:
ok so in jbang cli we would not call jdkmanger() but have our own instance configured from jbang config?

Reply:
Well, this `JavaUtil` class _is_ in the jbang CLI, so this code here is exactly where that instance using jbang config is created.

Reply:
NB: If you look a bit lower in the `createProvider()` method you see it references `config`, which is the jbang config. The missing part I mentioned in our chat is exactly that: tying together the config with the instance creation :-)